common/observer/event_dispatcher.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
should this file here? I think it should be in `common/extension`

Reply:
At first, I think so. But it can't be fixed about the problem of circular dependency, because of  extension package dependency on observer's interface and registry's interface, registry's implement dependency on extension package, observer's interface and registry's interface.